r/tests/testthat/test-dataset.R

```diff
@@ -417,6 +417,17 @@ test_that("filter() with is.na()", {
   )
 })
 
+test_that("filter() with is.na()", {
+  ds <- open_dataset(dataset_dir, partitioning = schema(part = uint8()))
+  expect_equivalent(
+    ds %>%
+      select(part, dbl) %>%
+      filter(!is.nan(dbl), part == 1) %>%
```

Comment:
I'm not sure this is a great test because the NaN you added looks like it is in `part == 2`